projects/angular-proto-firestore/src/app/core/services/firebase.service.ts

```diff
@@ -3,47 +3,61 @@ import { AngularFirestore, AngularFirestoreCollection } from 'angularfire2/fires
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 
-import { Book } from '@/core/proto';
+import { StoryList } from '@/core/proto';
 import { EncodingService } from './encoding.service';
-
-interface FirebaseElement {
-  proto: string;
+import { AngularFireAuth } from 'angularfire2/auth';
+import * as firebase from 'firebase/app';
+interface FirebaseElementStory {
+  story: string;
 }
 
 @Injectable()
 export class FirebaseService {
-  private protobin: AngularFirestoreCollection<FirebaseElement>;
+  isOnline: boolean;
+  private protobin: AngularFirestoreCollection<FirebaseElementStory>;
 
   constructor(
     private db: AngularFirestore,
     private encodingService: EncodingService,
+    private angularFireAuth: AngularFireAuth,
   ) {
-    this.protobin = this.db.collection('protobin');
+      this.angularFireAuth.authState.subscribe(userData => {
+      this.isOnline = !!userData;
+      });
+    this.protobin = this.db.collection('storyteller');
   }
 
-  getBook(): Observable<Book> {
-    return this.protobin
-      .doc('book')
-      .snapshotChanges()
-      .pipe(
-        map(action => {
-          const firebaseElement = action.payload.data() as FirebaseElement;
+  getstorylistAll(): Observable<StoryList[]> {
+    // this.shirts = this.shirtCollection.snapshotChanges().pipe(
+    //   map(actions => actions.map(a => {
+    //     const data = a.payload.doc.data() as Shirt;
+    //     const id = a.payload.doc.id;
+    //     return { id, ...data };
+    //   }))
+    return this.protobin.snapshotChanges().pipe(
+        map(action => action.map(a => {
+          const firebaseElement = a.payload.doc.data() as FirebaseElementStory;
+
           if (firebaseElement === undefined) {
             // Element not found
             return;
           }
-          return this.convertFirebaseElementToBook(firebaseElement);
-        })
-      );
+          console.log(firebaseElement['proto']);
```

Comment:
Never use `object['property']`, because it's weak typed way.
Strongly typed way is your friend. It prevents you from making mistakes like this one.
You use `FirebaseElementStory` with `story` child. But the data contains  'proto` only.
To get proto you need to change type to FirebaseElement instead of using `object['property']`.
Strongly typed way also provides code suggestions.